dbms/src/Storages/StorageDisaggregated.cpp

```diff
@@ -0,0 +1,270 @@
+}
+
+void StorageDisaggregated::setGRPCErrorMsg(const std::string & err)
+{
+    std::lock_guard<std::mutex> lock(err_msg_mu);
+    // Only record first err_msg.
+    if (err_msg.empty())
+    {
+        err_msg = err;
+    }
+}
+
+void StorageDisaggregated::buildReceiverStreams(const std::vector<RequestAndRegionIDs> & dispatch_reqs, unsigned num_streams, DAGPipeline & pipeline)
+{
+    tipb::ExchangeReceiver receiver;
+    for (const auto & dispatch_req : dispatch_reqs)
+    {
+        const ::mpp::TaskMeta & sender_task_meta = std::get<0>(dispatch_req)->meta();
+        receiver.add_encoded_task_meta(sender_task_meta.SerializeAsString());
+    }
+
+    const auto & column_infos = table_scan.getColumns();
+    for (const auto & column_info : column_infos)
+    {
+        auto * field_type = receiver.add_field_types();
+        auto tidb_column_info = TiDB::toTiDBColumnInfo(column_info);
+        *field_type = columnInfoToFieldType(tidb_column_info);
+    }
+
+    // ExchangeSender just use TableScan's executor_id, so exec summary will be merged to TableScan.
+    const auto & sender_target_task_meta = context.getDAGContext()->getMPPTaskMeta();
+    const String executor_id = table_scan.getTableScanExecutorID();
+    exchange_receiver = std::make_shared<ExchangeReceiver>(
+        std::make_shared<GRPCReceiverContext>(
+            receiver,
+            sender_target_task_meta,
+            context.getTMTContext().getKVCluster(),
+            context.getTMTContext().getMPPTaskManager(),
+            context.getSettingsRef().enable_local_tunnel,
+            context.getSettingsRef().enable_async_grpc_client),
+        receiver.encoded_task_meta_size(),
+        num_streams,
+        log->identifier(),
+        executor_id,
+        /*fine_grained_shuffle_stream_count=*/0,
+        /*is_receiver_for_tiflash_storage=*/true);
+
+    // MPPTask::receiver_set will record this ExchangeReceiver, so can cancel it in ReceiverSet::cancel().
+    context.getDAGContext()->setDisaggregatedComputeExchangeReceiver(executor_id, exchange_receiver);
+
+    // We can use PhysicalExchange::transform() to build InputStream after
+    // DAGQueryBlockInterpreter is deprecated to avoid duplicated code here.
+    const String extra_info = "disaggregated compute node exchange receiver";
+    for (size_t i = 0; i < num_streams; ++i)
+    {
+        BlockInputStreamPtr stream = std::make_shared<ExchangeReceiverInputStream>(exchange_receiver,
+                                                                                   log->identifier(),
+                                                                                   executor_id,
+                                                                                   /*stream_id=*/0);
+        stream->setExtraInfo(extra_info);
+        pipeline.streams.push_back(stream);
+    }
+
+    auto & table_scan_io_input_streams = context.getDAGContext()->getInBoundIOInputStreamsMap()[table_scan.getTableScanExecutorID()];
```

Comment:
Because TableScanStatistics assumes that table_scan_io_input_stream must be CopreadStream or LocalScanStream, it is necessary to modify TableScanStatistics to consider ExchangeReceiverStream

Reply:
Done

Reply:
It seems to have not been modified